src/magic_spell_effect.cpp

```diff
@@ -0,0 +1,461 @@
+                    targets.emplace( potential_target );
+                } else {
+                    break;
+                }
+            }
+        }
+        for( const tripoint &ccwise_line_pt : ccwise_line ) {
+            std::vector<tripoint> temp_line = line_to( start_line_pt, ccwise_line_pt );
+            for( const tripoint &potential_target : temp_line ) {
+                if( ignore_walls || g->m.passable( potential_target ) ) {
+                    targets.emplace( potential_target );
+                } else {
+                    break;
+                }
+            }
+        }
+    }
+
+    targets.erase( source );
+
+    return targets;
+}
+
+// spells do not reduce in damage the further away from the epicenter the targets are
+// rather they do their full damage in the entire area of effect
+static std::set<tripoint> spell_effect_area( const spell &sp, const tripoint &source,
+        const tripoint &target,
+        std::function<std::set<tripoint>( const spell &, const tripoint &, const tripoint &, const int, const bool )>
+        aoe_func, bool ignore_walls = false )
+{
+    std::set<tripoint> targets = { target }; // initialize with epicenter
+    if( sp.aoe() <= 1 ) {
+        return targets;
+    }
+
+    const int aoe_radius = sp.aoe();
+    targets = aoe_func( sp, source, target, aoe_radius, ignore_walls );
+
+    for( const tripoint &p : targets ) {
+        if( !sp.is_valid_target( p ) ) {
+            targets.erase( p );
+        }
+    }
+
+    // Draw the explosion
+    std::map<tripoint, nc_color> explosion_colors;
+    for( auto &pt : targets ) {
+        explosion_colors[pt] = sp.damage_type_color();
+    }
+
+    explosion_handler::draw_custom_explosion( g->u.pos(), explosion_colors );
+    return targets;
+}
+
+static void add_effect_to_target( const tripoint &target, const spell &sp )
+{
+    const int dur_moves = sp.duration();
+    const time_duration dur_td = 1_turns * dur_moves / 100;
+
+    Creature *const critter = g->critter_at<Creature>( target );
+    Character *const guy = g->critter_at<Character>( target );
+    efftype_id spell_effect( sp.effect_data() );
+    bool bodypart_effected = false;
+
+    if( guy ) {
+        for( const body_part bp : all_body_parts ) {
+            if( sp.bp_is_affected( bp ) ) {
+                guy->add_effect( spell_effect, dur_td, bp, sp.has_flag( "PERMANENT" ) );
+                bodypart_effected = true;
+            }
+        }
+    }
+    if( !bodypart_effected ) {
+        critter->add_effect( spell_effect, dur_td, num_bp );
+    }
+}
+
+static void damage_targets( const spell &sp, std::set<tripoint> targets )
+{
+    for( const tripoint target : targets ) {
+        Creature *const cr = g->critter_at<Creature>( target );
+        if( !cr ) {
+            continue;
+        }
+
+        projectile bolt;
+        bolt.impact = sp.get_damage_instance();
+        bolt.proj_effects.emplace( "magic" );
```

Comment:
This doesn't add bolt.speed, leaving it at zero and making magic quite easy to dodge. I tried with bolt.speed = 10000 and it fixes AOE spells not hitting all targets, what's Vorm's complaining about. But maybe it is intended. @KorGgenT

Reply:
_that's_ what fixes it? dang! make a pr =D